dataikuapi/dss/ml.py

```diff
@@ -308,50 +332,679 @@ def set_metric(self, metric=None, custom_metric=None, custom_metric_greater_is_b
+
+    def set_validation_n_folds(self, n_folds):
+        assert isinstance(n_folds, int)
+        if self._raw_settings["mode"] not in {"KFOLD", "TIME_SERIES_KFOLD"}:
+            warnings.warn("\'nFolds\' parameter will be ignored in {} mode".format(self._raw_settings["mode"]))
+        self._raw_settings["nFolds"] = n_folds
+
+    def set_validation_split_ratio(self, split_ratio):
+        assert isinstance(split_ratio, float)
+        if self._raw_settings["mode"] not in {"SHUFFLE", "TIME_SERIES_SINGLE_SPLIT"}:
+            warnings.warn("\'splitRatio\' parameter will be ignored in {} mode".format(self._raw_settings["mode"]))
+        self._raw_settings["splitRatio"] = split_ratio
+
+    def set_validation_n_shuffle_iter(self, n_shuffle_iter):
+        assert isinstance(n_shuffle_iter, int)
+        if self._raw_settings["mode"] != "SHUFFLE":
+            warnings.warn("\'shuffleIterations\' parameter will be ignored in {} mode".format(self._raw_settings["mode"]))
+        self._raw_settings["shuffleIterations"] = n_shuffle_iter
+
+    def set_execution_timeout(self, timeout):
+        assert isinstance(timeout, int)
+        self._raw_settings["timeout"] = timeout
+
+    def set_execution_n_iter(self, n_iter):
+        assert isinstance(n_iter, int)
+        if self._raw_settings["strategy"] == "GRID":
+            self._raw_settings["nIter"] = n_iter
+        else:
+            self._raw_settings["nIterRandom"] = n_iter
+
+    def set_grid_search_shuffling(self, shuffled, seed=None):
+        assert isinstance(shuffled, bool)
+        if seed is not None:
+            assert isinstance(seed, int)
+            self._raw_settings["seed"] = seed
+        self._raw_settings["randomized"] = shuffled
+        if self._raw_settings["strategy"] != "GRID":
+            warnings.warn("\'randomized\' and \'seed\' parameters will be ignored in {} strategy".format(self._raw_settings["strategy"]))
+
+    def set_search_distribution(self, distributed, n_containers):
+        assert isinstance(distributed, bool)
+        if n_containers is not None:
+            assert isinstance(n_containers, int)
+            self._raw_settings["nContainers"] = n_containers
+        self._raw_settings["distributed"] = distributed
+
+    def set_n_jobs(self, n_jobs):
+        assert isinstance(n_jobs, int) and n_jobs > 0
+        self._raw_settings["nJobs"] = n_jobs
+
+
+class HyperparameterSettings(object):
+
+    def __init__(self, name, algo_settings):
+        self.name = name
+        self._algo_settings = algo_settings
+
+
+class NumericalHyperparameterSettings(HyperparameterSettings):
+
+    def __repr__(self):
+        clean_dict = self._algo_settings._get_active_numerical_hyperparam_dict(self.name)
+        return self.__class__.__name__ + "(hyperparameter=\"{}\", settings={})".format(self.name, json.dumps(clean_dict, indent=4))
+
+    __str__ = __repr__
+
+    def set_explicit_values(self, values, mode_as_explicit=True):
```

Comment:
No strong opinion about the 2nd parameter.

A point about the name though (and same for `set_mode_as_*` and `_set_hyperparameter_mode_as_*`): in the rest of the code, `_as_*` is used for casting / type. Not sure we want to divert from this, maybe we can find an alternative name? `set_explicit_mode`, `set_mode_to_explicit`?

Reply:
Regarding `*_as_*`, I didn't pay attention but I agree on enforcing a consistency. Since `set_mode_to_explicit` is already used as a method name, I suggested `mode_becomes_explicit` and `mode_becomes_range`.

Reply:
Sound a bit weird, doesn't it? I don't have a problem with using the same name as a method if it does the same thing / calls the method (which it seems to do internally).

On the other hand, don't we want to use the same rationale as the validation mode and grid search method, and simplify to a single method for explicit + values, one for range + limits, etc? (And potentially keep gettable+settable individual properties on the side if we deem it useful).